src/main/java/com/adyen/util/StringUtil.java

```diff
@@ -0,0 +1,85 @@
+package com.adyen.util;
+
+import java.util.Arrays;
+
+public final class StringUtil {
+
+    private static final char MASK_CHAR = '*';
+    private static final int MASK_FIXED_BLOCK_SIZE = 3;
+    private static final String MASKED_BLOCK;
+    private static final int CARD_NUMBER_UNMASK_FIRST_BLOCK_SIZE = 6;
+    private static final int CARD_NUMBER_UNMASK_LAST_BLOCK_SIZE = 4;
+
+    static {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < MASK_FIXED_BLOCK_SIZE; i++) {
+            sb.append(MASK_CHAR);
+        }
+        MASKED_BLOCK = sb.toString();
+    }
+
+    private StringUtil() {
+    }
+
+    /**
+     * Masks the given string with a fixed size string of mask characters
+     * Return an empty string in case input is null or empty
+     *
+     * @param strToMask string to mask
+     * @return masked string
+     */
+    public static String maskWithFixedSize(String strToMask) {
```

Comment:
It returns a fixed string in case the string is not empty.
I think it comes down to the definition of masking :)  
It still kind of "masks" the input string with a fixed string